corehq/apps/app_manager/tests/test_suite_session_endpoints.py

```diff
@@ -0,0 +1,91 @@
+from django.test import SimpleTestCase
+
+from corehq.apps.app_manager.xform_builder import XFormBuilder
+from corehq.util.test_utils import flag_enabled
+
+from .app_factory import AppFactory
+from .util import (
+    TestXmlMixin,
+    patch_get_xform_resource_overrides,
+    patch_validate_xform,
+)
+
+
+@patch_validate_xform()
+@patch_get_xform_resource_overrides()
+@flag_enabled('SESSION_ENDPOINTS')
+class SessionEndpointTests(SimpleTestCase, TestXmlMixin):
```

Comment:
@shubham1g5 this test suite contains a few examples of generated suite files - a registration form, followup form, and a case list.  Do you have any more complex examples that would be useful to add?  Perhaps weirder scenarios like child case lists, for example.  I'm not confident all edge-cases are handled appropriately.

Reply:
This looks great, I agree that it would be nice to add an example for parent child workflow. [This form](https://staging.commcarehq.org/a/another-upstream/apps/view/b1e763d627164bccb1359585e0a5b7f7/form/05e6f3afe12c4048a7b26bd46963fd66/source/#form/the_show_is_in_case_name) in particular looks like a good example to cover. The suite file for it defines the form entry with the datums - 

````
 <session>
      <datum id="parent_id" nodeset="instance('casedb')/casedb/case[@case_type='song'][@status='open']" value="./@case_id" detail-select="m2_case_short"/>
      <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='show'][@status='open'][index/parent=instance('commcaresession')/session/data/parent_id]" value="./@case_id" detail-select="m3_case_short" detail-confirm="m3_case_long"/>
    </session>
````

so the session endpoint should define 2 arguments for this aka `$parent_id` and `$case_id`

Reply:
Great, just added a couple of examples for that.   Let me know if anything else comes up that should be tested explicitly.